NER evaluation on CoNLL03

Michael-Z-J opened the pull request:
added data preparation, updated version of main, evaluation file, and output files of evaluation

Aatlantise commented on evaluation_only.py:
This is a nice working script, but ideally we would implement this by adding a flag, maybe something like `--train` and `--eval` in `main.py`.

Aatlantise commented on main_fine_tune_conll.py:
Same deal as above--this creates many lines of duplicate code with `main.py`. We want to avoid creating duplicate lines of code! I think there's a way to implement this by using argparse flags. Also, `conll` might be confusing bc there are many datasets under the CoNLL umbrella!

Aatlantise commented on prep_conll.py:
Thanks for the CoNLL preprocessing code! I think it might be more beneficial to combine this with the existing `prep-pr-data.py`.

Aatlantise commented on run_eval_only.sh:
I think it's in general a good practice to not include log files and user-specific sbatch files in git commits.

Aatlantise commented on eval.py:
Was there a need for a new function? It would be best if existing lines of code can be adapted. Feel free to just change them since we're not running anything on the old code!

Aatlantise reviewed (changes requested):
Thanks Michael for implementing this!

Overall, in addition to my comments from today's weekly, watch out for:
* duplicate lines of code
* log files and user-specific scripts that may clutter the codebase.
Thanks!

Michael-Z-J commented:
Deleted redundant files
To prepare data run: python prep-data-pr.py with flag --dataset_name conll2003
Changed to parentheses to prepare labels e.g. target:"(Germany:LOC) (Britain:LOC)"

added flags to main.py
NER_eval_flag: to toggle if it runs NER or not
dataset_name: string for data file name (the jsonl file)
evaluate_only: only run the evaluation and return

Aatlantise commented on main.py:
```diff
@@ -361,5 +398,9 @@ def generate(ckpt: Union[str, None], model, input_dataset, tokenizer, batch_size
     parser = argparse.ArgumentParser()
     parser.add_argument("--model_name_or_path", type=str, default="t5-base")
     parser.add_argument("--output_dir", type=str, default="./outputs")
+    parser.add_argument("--NER_eval_flag", action="store_true")
```
Do you think we'd be able to use task name rather than task flags? That way, when we generalize to 10+ tasks, we don't need that many parser arguments.

Aatlantise commented on main.py:
```diff
@@ -327,7 +343,28 @@ def run(
+        print("NER evaluation running: ")
+        run_NER(dataset_name, output_dir, eval_batch_size=32, max_seq_length=256)
+
+def run_NER(dataset_name: str, output_dir: str, eval_batch_size: int=32, max_seq_length: int =256):
```
Similarly, do you think this could be achieved by parametrizing `run()` with a `task_name` parameter?

Aatlantise reviewed (changes requested):
Thanks for cleaning up the pipeline! This looks great, but I still have some minor nits I'd like to pick :)

Michael-Z-J commented:
Added task name flags (removed task flag)
using object_generation_score
now using load_prepped_dataset instead of having custom paths
parameterised run_evaluation

Aatlantise reviewed (approved):
Thanks for the changes! Looks good to me.
